indra/llui/llfloater.cpp

```diff
@@ -2996,16 +2951,13 @@ void LLFloaterView::draw()
 	refresh();
 
 	// hide focused floater if in cycle mode, so that it can be drawn on top
-	LLFloater* focused_floater = getFocusedFloater();
-
-	if (mFocusCycleMode && focused_floater)
+	if (LLFloater* focused_floater = mFocusCycleMode ? getFocusedFloater() : nullptr)
```

Comment:
This looks ugly... a ternary inside if?

Reply:
Yes, this should work fine and is compact; however, we need to check our coding conventions.

Reply:
As for me, it's readable enough

But the more important thing is that `getFocusedFloater()` isn't beong called when `mFocusCycleMode` is false

Reply:
Ternary is harder and more confusing to read by definition, not debugable. Good rule of thumb is to use it for short checks only and to never nest it. You are nesting it inside another if.

> getFocusedFloater() isn't being called when mFocusCycleMode is false
Since getFocusedFloater is const, it changes nothing by definition, compiler should be smart enough to optimize and not call getFocusedFloater() before mFocusCycleMode. And if you wanted to be explicit about that, you could have done that without nesting a ternary.